lib/deployment_scheduler.rb

```diff
@@ -0,0 +1,96 @@
+class DeploymentScheduler
+  def initialize(definition_parser = DefinitionParser.new)
+    @definition_parser = definition_parser
+  end
+
+  def schedule
+    Deployment.all.each do |deployment|
+      begin
+        process(deployment)
+        deployment.update!(last_error: nil)
+      rescue Exception => e
+        deployment.update!(last_error: "#{e.class.name}: #{e.message}")
+      end
+    end
+  end
+
+  private
+  def process(deployment)
+    wanted_hostnames = Set.new(deployment.container_names)
```

Comment:
what does `wanted` means?

Reply:
Based on [this explanation](https://github.com/pathfinder-cm/pathfinder-mono/pull/43#issuecomment-600389796), `wanted` is containers which should be in Pathfinder based on deployment name and replica count.

Reply:
ok, a suggestion how about `desired_replica_hostnames` to make it consistent with our definition.

Reply:
Noted.